ts/nni_manager/core/nniTensorboardManager.ts

```diff
@@ -0,0 +1,219 @@
+            trialJobIdList.push(trialJobId);
+            trialLogDirectoryList.push(trialDataPath);
+        }));
+        this.log.info(`tensorboard: ${trialJobIdList} ${trialLogDirectoryList}`);
+        return this.startTensorboardTaskProcess(trialJobIdList, trialLogDirectoryList);
+    }
+
+    private async startTensorboardTaskProcess(trialJobIdList: string[], trialLogDirectoryList: string[]): Promise<TensorboardTaskDetail> {
+        const deferred = new Deferred<TensorboardTaskDetail>();
+        try{
+            const host = 'localhost';
+            const port = await getFreePort(host, 6006, 65535);
+            const command = this.getTensorboardStartCommand(trialJobIdList, trialLogDirectoryList, port);
+            this.log.info(`tensorboard start command: ${command}`);
+            const tensorboardProc: ChildProcess = getTunerProc(command, 'ignore', process.cwd(), process.env);
+            const tensorboardTask = new TensorboardTaskDetail(uniqueString(5), 'RUNNING', trialJobIdList, trialLogDirectoryList);
+            this.tensorboardTaskMap.set(tensorboardTask.id, tensorboardTask);
+            tensorboardTask.pid = tensorboardProc.pid;
+            tensorboardTask.port = `${port}`;
+            this.log.info(`tensorboard task id: ${tensorboardTask.id}`);
+            this.updateTensorboardTask(tensorboardTask.id);
+            deferred.resolve(tensorboardTask);
+        } catch (error) {
+            deferred.reject(error);
+        }
+        return deferred.promise
+    }
+
+    private getTensorboardStartCommand(trialJobIdList: string[], trialLogDirectoryList: string[], port: number): string {
+        if (this.tensorboardVersion === undefined) {
+            this.setTensorboardVersion();
+            if (this.tensorboardVersion === undefined) {
+                throw new Error(`Tensorboard may not installed, if you want to use tensorboard, please check if tensorboard installed.`);
+            }
+        }
+        if (trialJobIdList.length !== trialLogDirectoryList.length) {
+            throw new Error('trial list length does not match');
+        }
+        if (trialJobIdList.length === 0) {
+            throw new Error('trial list length is 0');
+        }
+        let logdirCmd = '--logdir';
+        if (this.tensorboardVersion >= '2.0') {
+            logdirCmd = '--bind_all --logdir_spec'
+        }
+        try {
+            const logRealPaths: string[] = [];
+            trialJobIdList.forEach((trialJobId, idx) => {
+                const realPath = fs.realpathSync(trialLogDirectoryList[idx]);
+                logRealPaths.push(`${trialJobId}:${realPath}`);
+            });
+            const command = `tensorboard ${logdirCmd}=${logRealPaths.join(',')} --port=${port}`;
+            return command;
+        } catch (error){
+            throw new Error(`${error.message}`);
+        }
+    }
+
+    private setTensorboardVersion(): void {
+        let command = 'python3 -m pip show tensorboard | grep Version:';
```

Comment:
`python -c 'import tensorboard ; print(tensorboard.__version__)'`

Reply:
yes, thank you